packages/playwright-core/src/server/chromium/crApp.ts

```diff
@@ -41,6 +41,9 @@ export async function syncLocalStorageWithSettings(page: Page, appName: string)
   const settings = await fs.promises.readFile(settingsFile, 'utf-8').catch(() => ('{}'));
   await page.addInitScript(
       `(${String((settings: any) => {
+        // iframes w/ snapshots, etc.
+        if (location && location.protocol === 'data:')
```

Comment:
When a blank snapshot iframe is created with data: url, we tried to inject localStorage